### DISCUSSION TITLE
Improve Function Editor UX / Resolve frame offset sync

### DISCUSSION THREAD

**Discussion by Al532** [opentoonz/opentoonz#6761] [Ideas]
Bug 1: When scrolling down between frames with the down arrow key, either in the xsheet or in the function editor, the function editor doesn't follow up  when the selection goes above the bottom, it starts to scroll only when it meets the xsheet bottom. This is problematic in the default Xsheet view, where the function editor hasn't the same height as the xsheet pannel. This seems however ok if the xsheet has the timeline display. 

Bug 2: When toggling Xsheet/Timeline while the current frame is not at the beginning (for the test purpose, select the frame 150), the function editor display will be completely offset from the current frame, and stays offset even when changing the current frame in the function editor (until we manually scroll and select the first frame in the function editor)

**Comment by RodneyBaker** [comment]
For issue 1:  You might dock the xsheet to the left of the Function Editor and make sure the option to add the header space is toggled on in xsheet > preferences.  This will ensure both Xsheet and Function Editor stay aligned and in sync.
